src/main/java/org/wildfly/security/auth/realm/ldap/UserPasswordCredentialLoader.java

```diff
@@ -81,7 +85,7 @@ EvidenceVerifier toEvidenceVerifier() {
             @Override
             public SupportLevel getEvidenceVerifySupport(final DirContextFactory contextFactory, final Class<? extends Evidence> evidenceType, final String algorithmName) throws RealmUnavailableException {
                 // If it is obtainable, it is verifiable.
-                return getCredentialAcquireSupport(contextFactory, PasswordCredential.class, null);
+                return getCredentialAcquireSupport(contextFactory, PasswordCredential.class, algorithmName);
```

Comment:
This bit of code needs the same treatment as lines 120-122 I think.  If the evidence type is ``PasswordGuessEvidence``, then return the result of ``getCredentialAcquireSupport`` for any algorithm (pass in ``null`` for the third argument).